inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/BitLengthFunction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Function;
+
+/**
+ * BitLengthFunction
+ * description: bit_length(string)
+ * - return NULL if the string is NULL.
+ * - return number of bits in string.
+ */
+@TransformFunction(names = {"bit_length"})
+public class BitLengthFunction implements ValueParser {
+
+    private final ValueParser stringParser;
+
+    public BitLengthFunction(Function expr) {
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObject = stringParser.parse(sourceData, rowIndex, context);
+        if (stringObject == null) {
+            return null;
+        }
+        return OperatorTools.parseString(stringObject).getBytes().length * 8;
```

Comment:
Different charset has different bit length.
getBytes will use runtime charset.
This function miss charset parameter.

Reply:
Thanks. According to your suggestion, the modification has been completed.